packages/pwa-kit-create-app/scripts/create-mobify-app.js

```diff
@@ -47,54 +47,355 @@ const tar = require('tar')
+    {
+        name: 'project.name',
+        validate: validProjectName,
+        message: 'What is the name of your Project?'
+    }
 ]
-const PUBLIC_PRESETS = [
-    RETAIL_REACT_APP_DEMO,
-    RETAIL_REACT_APP,
-    EXPRESS_MINIMAL,
-    TYPESCRIPT_MINIMAL
+
+const TYPESCRIPT_MINIMAL_QUESTIONS = [
+    {
+        name: 'project.name',
+        validate: validProjectName,
+        message: 'What is the name of your Project?'
+    }
 ]
-const PRESETS = PRIVATE_PRESETS.concat(PUBLIC_PRESETS)
 
-const DEFAULT_OUTPUT_DIR = p.join(process.cwd(), 'pwa-kit-starter-project')
+const RETAIL_REACT_APP_QUESTIONS = [
+    {
+        name: 'project.name',
+        validate: validProjectName,
+        message: 'What is the name of your Project?'
+    },
+    {
+        name: 'project.commerce.instanceUrl',
+        message: 'What is the URL for your Commerce Cloud instance?',
+        validate: validUrl
+    },
+    {
+        name: 'project.commerce.clientId',
+        message: 'What is your SLAS Client ID?',
+        validate: validClientId
+    },
+    {
+        name: 'project.commerce.siteId',
+        message: 'What is your Site ID in Business Manager?',
+        validate: validSiteId
+    },
+    {
+        name: 'project.commerce.organizationId',
+        message: 'What is your Commerce API organization ID in Business Manager?',
+        validate: validOrganizationId
+    },
+    {
+        name: 'project.commerce.shortCode',
+        message: 'What is your Commerce API short code in Business Manager?',
+        validate: validShortCode
+    }
+]
+
+// Project dictionary describing details and how the gerator should ask questions etc.
+const PRESETS = [
+    {
+        id: 'retail-react-app',
+        name: 'Retail React App',
+        description: `
+            Generate a project using custom settings by answering questions about a
+            B2C Commerce instance.
+    
+            Use this preset to connect to an existing instance, such as a sandbox.
+        `,
+        shortDescription: 'The Retail app using your own Commerce Cloud instance',
+        templateSource: {
+            type: TEMPLATE_SOURCE_NPM,
+            id: 'retail-react-app'
+        },
+        questions: [...EXTENSIBILITY_QUESTIONS, ...RETAIL_REACT_APP_QUESTIONS],
+        private: false
+    },
+    {
+        id: 'retail-react-app-demo',
+        name: 'Retail React App Demo',
+        description: `
+            Generate a project using the settings for a special B2C Commerce
+            instance that is used for demo purposes. No questions are asked.
+    
+            Use this preset to try out PWA Kit.
+        `,
+        shortDescription: 'The Retail app with demo Commerce Cloud instance',
+        templateSource: {
+            type: TEMPLATE_SOURCE_NPM,
+            id: 'retail-react-app'
+        },
+        questions: [...EXTENSIBILITY_QUESTIONS, ...RETAIL_REACT_APP_QUESTIONS],
+        answers: {
+            ['project.extend']: true,
```

Comment:
Minor: we can rely on the quotes.
```suggestion
            'project.extend': true,
```

Reply:
I personally like using brackets for properties that require brackets to get the value. Unless we have a lint rule for this,  lets just leave it, we can talk about making a rule on computed properties or when to use and no use brackets .